homeassistant/components/hyperion/light.py

```diff
@@ -346,6 +346,9 @@ def is_on(self) -> bool:
     @property
     def icon(self) -> str:
         """Return state specific icon."""
+        if not self._client.is_on():
+            return ICON_LIGHTBULB
+
```

Comment:
Three recommendations:
   * Work this into set_internal_state() instead (this is where the icons are set to date).
   * Don't use client.is_on() explicitly, call is_on() as that's the authority on whether it's on or off (and the contents of that method may change and we don't want to have to update it here).
   * Personally, I don't have an objection about including this unrelated change here -- but the Home Assistant core team will have an objection (based on experience). We will need their approval to get this merged in, so you'll need to break this into two PRs. Small PRs are also quicker to get reviewed/merged.